scripts/test_commands.sh

```diff
@@ -0,0 +1,86 @@
+# EXAMPLE:
+# ./scripts/sanity_check.sh -c IntegrationTestCluster -r us-east-2
+# -a $AWS_ACCESS_KEY_ID -s $AWS_SECRET_ACCESS_KEY -k MyFirstKeyPair
+# -p ~/Downloads/KeyPair.pem -i ec2-18-220-215-243.us-east-2.compute.amazonaws.com
+# -u PettitWesley -b master
+
+usage() {
+  echo "Usage: ${0}"
+  echo
+  echo "This script runs a series of ECS CLI commands on a fresh EC2 instance."
+  echo "It is designed to provide integration tests for the ECS CLI."
+  echo "However, it is called sanity check for a reason, this script simply"
+  echo "checks if the commands run exited with code 0."
+  echo "More detailed test output can be examined manually in ~/test_output.txt,"
+  echo "on the ec2 instance."
+  echo
+  echo "Required Arguments:"
+  echo "  -c  CLUSTER            A name for the cluster that will be created/used in the tests."
+  echo "  -r  REGION             Region to create AWS resources in."
+  echo "  -a  ACCESS KEY         The access key id for an AWS account."
+  echo "  -s  SECRET KEY         The secret key for an AWS account."
+  echo "  -k  KEY PAIR NAME      A keypair for the chosen region."
+  echo "  -p  KEY PAIR PATH      The path to the key pair needed to ssh into the given ec2 instance."
+  echo "  -i  INSTANCE URL       The public DNS for the ec2 instance created for testing."
+  echo "  -u  GITHUB USERNAME    The testers github username."
+  echo "  -b  BRANCH             The branch on github that will be tested."
+  echo "  -h                     Display this help message"
+}
+
+# ARGS: cluster c, region r, access a, secret s, keypair k , keypath p, instance_url i, gitname u, branch b
+while getopts ":c:r:a:s:k:p:i:u:b:" opt; do
+	case ${opt} in
+		c)
+			cluster="${OPTARG}"
+			;;
+		r)
+			region="${OPTARG}"
+			;;
+		a)
+			access="${OPTARG}"
+			;;
+		s)
+			secret="${OPTARG}"
+			;;
+		k)
+			keypair="${OPTARG}"
+			;;
+		p)
+			keypath="${OPTARG}"
+			;;
+		i)
+			instance_url="${OPTARG}"
+			;;
+		u)
+			gitname="${OPTARG}"
+			;;
+		b)
+			branch="${OPTARG}"
+			;;
+		\?)
+			echo "Invalid option -${OPTARG}" >&2
+			usage
+			exit 1
+			;;
+		:)
+			echo "Option -${OPTARG} requires an argument." >&2
+			usage
+			exit 1
+			;;
+		h)
+			usage
+			exit 0
+			;;
+	esac
+done
+
+if [ -z "${instance_url}" ]; then
+	usage
+	exit 1
+fi
+
+scp -i $keypath $(dirname "${0}")/run_commands.sh "ec2-user@${instance_url}":~/
```

Comment:
Could we use `rsync` here?

Reply:
(in case, for example, someone has an existing instance that they are unfrugally keeping up for testing, and have already copied the test script there at some point. Rsync in this scenario would be a tad faster, i believe)

Reply:
That's true, `rsync` would probably be slightly faster in some cases. But I think `scp` is fine enough.